enterprise/backend/src/metabase_enterprise/transforms/api.clj

```diff
@@ -243,13 +241,8 @@
   (let [transform (api/check-404 (t2/select-one :model/Transform id))
         _         (check-feature-enabled! transform)
         start-promise (promise)]
-    (if (transforms.util/python-transform? transform)
-      (u.jvm/in-virtual-thread*
-       (transforms-python.execute/execute-python-transform! transform {:start-promise start-promise
-                                                                       :run-method :manual}))
-      (u.jvm/in-virtual-thread*
-       (transforms.execute/run-mbql-transform! transform {:start-promise start-promise
-                                                          :run-method :manual})))
+    (transforms.i/execute! transform {:start-promise start-promise
```

Comment:
I would personally keep the thread on the outside, its independent from the implementation afaict.